vnext/Microsoft.ReactNative/Modules/PaperUIManagerModule.cpp

```diff
@@ -0,0 +1,793 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+#include "pch.h"
+#include <DynamicReader.h>
+#include <DynamicWriter.h>
+#include <IReactContext.h>
+#include <IReactRootView.h>
+#include <Modules/PaperUIManagerModule.h>
+#include <Modules\NativeUIManager.h>
+#include <Views/ViewManager.h>
+#include <XamlUtils.h>
+#include "ShadowNodeBase.h"
+#include "Unicode.h"
+#include "XamlUIService.h"
+
+namespace Microsoft::ReactNative {
+
+struct ViewAtIndex {
+  int64_t tag;
+  int64_t index;
+  static bool ViewAtIndexCompare(
+      const std::shared_ptr<ViewAtIndex> &x,
+      const std::shared_ptr<ViewAtIndex> &y) noexcept {
+    return (x->index < y->index);
+  }
+};
+
+std::vector<int64_t> PopulateIntVecFromDynamicArray(React::JSValueArray &arr) {
+  std::vector<int64_t> vec(arr.empty() ? 0 : arr.size());
+
+  if (arr.empty())
+    return vec;
+
+  auto i = 0;
+  for (auto &val : arr) {
+    vec[i] = val.AsInt64();
```

Comment:
Do we need to do any checks here that it is a number?  technically arrray's in JSValue can be contain mixed types.

Reply:
There are a lot of places in this change where there are not enough checks on the JSValues.  But I'm trying to keep the logic as much the same as before the change, since the change already has enough churn for one PR.